src/analysis/seminorm.lean

```diff
@@ -45,51 +46,196 @@ open_locale big_operators nnreal pointwise topological_space
 
 variables {R R' 𝕜 E F G ι : Type*}
 
-/-- A seminorm on an add_group `G` is a function A function `f : G → ℝ` that preserves zero, takes
+/-- A seminorm on an additive group `G` is a function `f : G → ℝ` that preserves zero, takes
 nonnegative values, is subadditive and such that `f (-x) = f x` for all `x ∈ G`. -/
 structure add_group_seminorm (G : Type*) [add_group G]
   extends zero_hom G ℝ :=
 (nonneg' : ∀ r, 0 ≤ to_fun r)
 (add_le' : ∀ r s, to_fun (r + s) ≤ to_fun r + to_fun s)
 (neg' : ∀ r, to_fun (- r) = to_fun r)
 
-attribute [nolint doc_blame] add_group_seminorm.to_zero_hom
+/-- A seminorm on a group `G` is a function `f : G → ℝ` that preserves zero, takes nonnegative
+values, is submultiplicative and such that `f x⁻¹ = f x` for all `x`. -/
+@[to_additive] structure group_seminorm (G : Type*) [group G] :=
+(to_fun : G → ℝ)
+(map_one' : to_fun 1 = 0)
+(nonneg' : ∀ x, 0 ≤ to_fun x)
+(mul_le' : ∀ x y, to_fun (x * y) ≤ to_fun x + to_fun y)
+(inv' : ∀ x, to_fun x⁻¹ = to_fun x)
 
-namespace add_group_seminorm
+attribute [nolint doc_blame] add_group_seminorm.to_zero_hom
```

Comment:
Seems like a good opportunity to add this doc!

Reply:
See #2409

Reply:
I am particularly unwilling to add this docstring because:
* It won't show up in the docs.
* `zero_hom` is mathematically irrelevant.

Reply:
Hmm, yeah you're right, it does show up.